### PR TITLE
Updated ldap configuration with active directory support

### PR DESCRIPTION
**Pull request by SachinKSingh28**
This PR includes the changes made in #2 
Updates :
- Using the static group configuration for AD as well.
- Added environment variable **LDAP_AUTH_PROTOCOL** to control the ldap protocol used. Which allows you to use ldap or ldaps.
- Added validation for custom roles.  
 - Custom roles should not be empty otherwise the security.xml is corrupted.
 - There must not be added if present. 
 - Categorized configuration into managed and unmanaged. This is to allow us to be able to make changes through UI which persists across restarts.

### CODE REVIEW

File: resources/nexus.sh [hunk 1]
```diff
@@ -6,67 +6,147 @@ echo "$(date) - LDAP Enabled: ${LDAP_ENABLED}"
 
 # Copy config files.
 mkdir -p ${NEXUS_HOME}conf
-cp -R /resources/* ${NEXUS_HOME}conf
+cp -R /resources/conf/managed/* ${NEXUS_HOME}conf
+cp -R -n /resources/conf/unmanaged/* ${NEXUS_HOME}conf
```
Comment (nickdgriffin) [inline resources/nexus.sh]: I think it'd be useful to have a comment near this block explaining the difference between the managed and unmanaged files please.

**Comment by nickdgriffin** [comment]
Looks good, only two minor things (one commented against the code) and the other being can LDAP_AUTH_PROTOCOL be added into the ENV block in the Dockerfile please?

Once those two changes are in I'm happy to merge this.

**Comment by SachinKSingh28** [comment]
Updated the Dockerfile and added the comments.

**Comment by nickdgriffin** [comment]
LGTM.

**Pull request merged**